static/app/utils/profiling/gl/utils.ts

```diff
@@ -624,3 +641,52 @@ export function computeHighlightedBounds(
 
   return null;
 }
+
+export function computeConfigViewWithStategy(
+  strategy: 'min' | 'exact',
+  view: Rect,
+  frame: Rect
+): Rect {
+  if (strategy === 'exact') {
+    return frame.withHeight(view.height);
+  }
+
+  if (strategy === 'min') {
+    if (view.width < frame.width) {
+      // If view width < frame width, we need to zoom out,
+      // so the condition = exact
+      return frame.withHeight(view.width);
+    }
+
+    if (view.containsRect(frame)) {
+      // If frame is in the frame, do nothing
+      return view;
+    }
+
+    let offset = view;
+    if (frame.left < view.left) {
+      // If frame is to the left of the view, translate it left
+      // to frame.x so that start of the frame is in the view
+      offset = offset.withX(frame.x);
+    }
+    if (offset.overlapsRight(frame) || frame.right > view.right) {
+      // If frame is to the right of the view, translate it right
+      // by the difference between the right edge of the frame and the view
+      offset = view.withX(offset.x + frame.right - offset.right);
+    }
```

Comment:
Can you explain what the intent here is? I'm not sure I understand what this does.